### PR TITLE
Added for getting a user attribute

### PR DESCRIPTION
**Pull request by RKathees**

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 1]
```diff
@@ -615,4 +622,41 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
...
+    /**
+     * Get a screen value from the user attributes.
+     *
+     * @param smsOTPParameters the sms OTP Parameters
+     * @param userRealm        the user Realm
+     * @param username         the username
+     * @return the screen attribute
+     * @throws UserStoreException
+     */
+    public String getScreenAttribute(Map<String, String> smsOTPParameters, UserRealm userRealm, String username)
```
Comment (malakasilva) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Can we make this method private?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 2]
```diff
@@ -615,4 +622,41 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
...
+            throws UserStoreException {
+        String screenUserAttributeParam;
+        String screenUserAttributeValue = null;
+        String screenValue = null;
+        int noOfDigits = 0;
+
+        screenUserAttributeParam = smsOTPParameters.get(SMSOTPConstants.SCREEN_USER_ATTRIBUTE);
+        if (screenUserAttributeParam != null) {
```
Comment (malakasilva) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Why the same condition checked two times?

Comment (RKathees) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: screenUserAttributeParam is param in authentication xml. screenUserAttributeValue is value of claim

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 1 above)
Comment (RKathees) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: We will need to extend this class and override the method when we build a customizable connector.

**Pull request merged**